### PR TITLE
[GEOT-6588] : SortBy element not encoded with FESConfiguration

### PR DESCRIPTION
**Pull request by eropartz**
<Include a few sentences describing the overall goals for this Pull Request>

Fix for [GEOT-6558](https://osgeo-org.atlassian.net/browse/GEOT-6588) . Element SortBy is not encoded in getFeature request (WFS 2.0)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by eropartz** [comment]
JIRA ticket : https://osgeo-org.atlassian.net/browse/GEOT-6588

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortByTypeBinding.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *          &lt;xsd:element maxOccurs="unbounded" name="SortProperty" type="ogc:SortPropertyType"/&gt;
+ *      &lt;/xsd:sequence&gt;
+ *  &lt;/xsd:complexType&gt;
+ *
+ *          </code>
+ * </pre>
+ *
+ * @generated
+ */
+public class SortByTypeBinding extends org.geotools.filter.v1_1.SortByTypeBinding {
```
Comment (ianturton) [inline modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortByTypeBinding.java:42]: Should this not extend a V2_0 class?

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/FESConfiguration.java [hunk 2]
```diff
@@ -18,9 +18,7 @@
 
 import net.opengis.fes20.Fes20Factory;
 import org.geotools.filter.FilterFactoryImpl;
-import org.geotools.filter.v1_1.SortByTypeBinding;
 import org.geotools.filter.v1_1.SortOrderTypeBinding;
```
Comment (ianturton) [inline modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/FESConfiguration.java:21]: Is this still needed? and if it is should it be?

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortPropertyTypeBinding.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ *          &lt;xsd:element minOccurs="0" name="SortOrder" type="ogc:SortOrderType"/&gt;
+ *      &lt;/xsd:sequence&gt;
+ *  &lt;/xsd:complexType&gt;
+ *
+ *          </code>
+ * </pre>
+ *
+ * @generated
+ */
+public class SortPropertyTypeBinding extends org.geotools.filter.v1_1.SortPropertyTypeBinding {
```
Comment (ianturton) [inline modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortPropertyTypeBinding.java:44]: Again this should probably extend a V2_0 class

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortByTypeBinding.java (hunk 1 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/SortByTypeBinding.java:42]: There is some code shared beetwen v1 and v2 ( parsing method). This inheritance is used to override the target, the type and the getProperty

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/FESConfiguration.java (hunk 2 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/FESConfiguration.java:21]: The code is identical between V1 and V2, so I used the class v1_1.SortOrderTypeBinding. If you prefer, I can create an empty class  v2.SortOrderTypeBinding which extends v1_1.SortOrderTypeBinding

Comment (ianturton) [reply modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/FESConfiguration.java:21]: No if they are identical then no problem, just caught my eye and have seen similar issues with this sort of mixing.

**Pull request merged**